packages/medusa/src/api/admin/products/[id]/route.ts

```diff
@@ -38,6 +38,16 @@ export const POST = async (
 ) => {
   const { additional_data, ...update } = req.validatedBody
 
+  /**
+   * Check if the product exists with the id or not before calling the workflow.
+   */
+  if (!(await refetchEntity("product", req.params.id, req.scope, ["id"]))) {
```

Comment:
NIT: Might be a bit cleaner if we move the refetch to not be inline with the `if`

Reply:
Sure